File: template-method/template-method.test.ts
import { PostAccountReqBody } from "./post-account-req-body";
import { PostAccountController } from "./post-account.controller";

test('Test Template Method', () => {
    const controller = new PostAccountController();
    const response = controller.createAccount(new PostAccountReqBody("12345", "123"));
    expect(response.status).toEqual(201);
});

File: go/template_method/template_method_test.go
package template_method

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestTemplateMethod(t *testing.T) {
	t.Run("Test Template Method", func(t *testing.T) {
		controller := NewPostAccountController[PostAccountReqBody, PostAccountResBody, string]()
		reqBody, err := NewPostAccountReqBody("12345", "123")
		if err != nil {
			t.Error()
		}
		response := controller.CreateAccount(*reqBody)

		assert.Equal(t, 201, response.Status)
	})
}
